projects/vdk-core/src/vdk/internal/builtin_plugins/run/cli_run.py

```diff
@@ -120,6 +120,26 @@ def __warn_on_python_version_disparity(
                         """
                     )
 
+    def __log_exec_result(self, execution_result):
+        # On some platforms, if the size of a string is too large, the
+        # logging module starts throwing OSError: [Errno 40] Message too long,
+        # so it is safer if we split large strings into smaller chunks.
+        string_exec_result = str(execution_result)
+        if len(string_exec_result) > 5000:
+            temp_exec_result = json.loads(string_exec_result)
```

Comment:
what if string_exec_result is not proper json ?

Reply:
Refactoring is out of scope for this PR. I'm just copying existing code to a new function. We can open a ticket to take a look at this at a later time.

Reply:
This is a new code as far as I can see. Moved from there? 
The code may fail. What are the guarantees that `json.loads(string_exec_result)` would not raise an exception . Logging  should not cause job to fail. 


Also we generally do not do refactoring dedicated stories unless they are big enough. Refactoring should be considered as a task as part of the implementation

Reply:
Please look at the diff again

![Screenshot 2023-11-14 at 16 36 43](https://github.com/vmware/versatile-data-kit/assets/91800778/0fdc4ecd-7e0a-4c4a-9216-a50b87a5dc20)

Reply:
To address your other points:

> Also we generally do not do refactoring dedicated stories unless they are big enough. Refactoring should be considered as a task as part of the implementation

The task is to put logging of the execution result behind a config option. I'm not about to introduce potentially breaking changes as part of this task. We can do it as part of another story and it's completely fine.

> The code may fail. What are the guarantees that json.loads(string_exec_result) would not raise an exception . Logging should not cause job to fail.

`str()` will try to call the object's `__str__()` method. If it's not implemented, it will fall back to `repr()` which calls the object's `__repr__()` method. In our case, we're calling `str()` on an `ExecutionResult` object.

Digging into the code a little, we can see it has a `__repr__` method. https://github.com/vmware/versatile-data-kit/blob/main/projects/vdk-core/src/vdk/internal/builtin_plugins/run/execution_results.py#L127

And the `__repr__` method handles the exact case you're referring to and falls back to the python pretty printer. https://github.com/vmware/versatile-data-kit/blob/main/projects/vdk-core/src/vdk/internal/builtin_plugins/run/execution_results.py#L164

It seems to me it will produce some kind of json if possible. https://docs.python.org/3/library/pprint.html#prettyprinter-objects

We're guaranteed to get an ExecutionResult object every time, because exceptions are handled in the `run` function. https://github.com/vmware/versatile-data-kit/blob/main/projects/vdk-core/src/vdk/internal/builtin_plugins/run/data_job.py#L327

Reply:
> We're guaranteed to get an ExecutionResult object every time, because exceptions are handled in the `run` function. https://github.com/vmware/versatile-data-kit/blob/main/projects/vdk-core/src/vdk/internal/builtin_plugins/run/data_job.py#L327

For now. But by contract str and repr are not guaranteed to be json parseable. So if someone changes their implementation , that would break. Though one can hope and expect tests would catch this. 

But you are correct. As this existing code and not newly introduced one is subject to separate refactoring (if we decide to do it).